master/internal/db/postgres_users.go

```diff
@@ -317,17 +317,19 @@ func (db *PgDB) AuthTokenKeypair() (*model.AuthTokenKeypair, error) {
 }
 
 // UpdateUserSetting updates user setting.
-func UpdateUserSetting(setting *model.UserWebSetting) error {
-	if len(setting.Value) == 0 {
-		_, err := Bun().NewDelete().Model(setting).Where(
-			"user_id = ?", setting.UserID).Where(
-			"storage_path = ?", setting.StoragePath).Where(
-			"key = ?", setting.Key).Exec(context.TODO())
-		return err
+func UpdateUserSetting(settings []*model.UserWebSetting) error {
+	var err error
+	for _, setting := range settings {
+		if len(setting.Value) == 0 {
+			_, err = Bun().NewDelete().Model(setting).Where(
```

Comment:
non-blocking: not really a part of this pr but
would a single update work here? or wait would the CONFLICT clause handle this? 🤔

Reply:
Single update doesn't work because the values you're setting aren't all the same.